File: nysol_python/mcmd/methods/convTex/db/smp/marff2csv.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='marff2csv'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.arff'
data['text']='''
@RELATION       customerPurchaseData
@ATTRIBUTE      customer    string
@ATTRIBUTE      date    date yyyyMMdd
@ATTRIBUTE      quantity    numeric
@ATTRIBUTE      amount    numeric
@ATTRIBUTE      item    {A,B}
@DATA
No.1,20081201,1,10,A
No.2,20081202,2,20,A
No.3,20081203,3,30,A
No.4,20081201,4,40,B
No.5,20081203,5,50,B
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
arff形式の顧客購買データをcsv形式のデータへ変換する。
'''
script['sh_code']='''
marff2csv i=dat1.arff  o=rsl1.csv
'''
script['py_code']='''
nm.marff2csv(i="dat1.arff", o="rsl1.csv").run()
'''
script['odatas']='rsl1.csv'
db['scripts'].append(script)

File: olddoc/mcmd/en/examples/marff2csv.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.arff","w"){|fpw| fpw.write(
<<'EOF'
@RELATION       Customer Purchase Data

@ATTRIBUTE      Customer    string
@ATTRIBUTE      Date    date yyyymmdd
@ATTRIBUTE      Quantity    numeric
@ATTRIBUTE      Amount    numeric
@ATTRIBUTE      Product    {A,B}

@DATA
No.1,20081201,1,10,A
No.2,20081202,2,20,A
No.3,20081203,3,30,A
No.4,20081201,4,40,B
No.5,20081203,5,50,B
EOF
)}

############## Example 1
title="Basic Example"
comment=<<'EOF'
Convert customer purchasing data in arff format to csv format. 
EOF
scp=<<'EOF'
more dat1.arff
marff2csv i=dat1.arff  o=rsl1.csv
more rsl1.csv
EOF
run(scp,title,comment)

